src/components/TradeAssetSearch/hooks/useFetchAndUpsertCustomTokenMarketData.tsx

```diff
@@ -0,0 +1,37 @@
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import { marketData as marketDataSlice } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectIsCustomAsset } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+import { getTokenMarketData } from './useGetCustomTokenPriceQuery'
+
+// Custom hook to fetch and upsert market data for custom tokens
+export const useFetchAndUpsertCustomTokenMarketData = () => {
+  const dispatch = useDispatch()
+
+  const fetchAndUpsertCustomTokenMarketData = useCallback(
+    async (assetId: string) => {
```

Comment:
```suggestion
    async (assetId: AssetId) => {
```

Reply:
Good call, addressed in https://github.com/shapeshift/web/pull/7370/commits/45d970641b800d671d18b0b40d9f616e9afe0f6d